index.php

```diff
@@ -47,6 +47,56 @@ function gutenberg_register_scripts() {
 add_action( 'init', 'gutenberg_register_scripts' );
 
 /**
+ * Adds the filters to register additional links for the Gutenberg editor in
+ * the post/page screens.
+ *
+ * @since 0.1.0
+ */
+function gutenberg_add_edit_links_filters() {
+	// For hierarchical post types
+	add_filter( 'page_row_actions', 'gutenberg_add_edit_links', 10, 2 );
+	// For non-hierarchical post types
+	add_filter( 'post_row_actions', 'gutenberg_add_edit_links', 10, 2 );
+}
+add_action( 'admin_init', 'gutenberg_add_edit_links_filters' );
+
+/**
+ * Registers additional links in the post/page screens to edit any post/page in
+ * the Gutenberg editor.
+ *
+ * @since 0.1.0
+ */
+function gutenberg_add_edit_links( $actions, $post ) {
+	$can_edit_post = current_user_can( 'edit_post', $post->ID );
+	$title = _draft_or_post_title( $post->ID );
+
+	if ( $can_edit_post && 'trash' !== $post->post_status ) {
+		// Build the Gutenberg edit action.  See also:
+		// WP_Posts_List_Table::handle_row_actions()
+		$gutenberg_url = menu_page_url( 'gutenberg', false );
+		$gutenberg_action = sprintf(
+			'<a href="%s" aria-label="%s">%s</a>',
+			add_query_arg( 'post_id', $post->ID, $gutenberg_url ),
+			/* translators: %s: post title */
+			esc_attr( sprintf(
+				__( 'Edit &#8220;%s&#8221; in the Gutenberg editor', 'gutenberg' ),
+				$title
+			) ),
+			__( 'Gutenberg', 'gutenberg' )
+		);
+		// Insert the Gutenberg action immediately after the Edit action.
+		$edit_offset = array_search( 'edit', array_keys( $actions ), true );
+		$actions = array_merge(
+			array_slice( $actions, 0, $edit_offset + 1 ),
+			array( 'gutenberg hide-if-no-js' => $gutenberg_action ),
+			array_slice( $actions, $edit_offset + 1 )
+		);
```

Comment:
This is a bit annoying, but `$actions` is an associative array and we need to insert a key into the middle of it.  Technique from http://stackoverflow.com/a/9847709/106302.